cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -348,7 +348,23 @@ if(PARQUET_BUILD_BENCHMARKS)
 endif()
 
 ## Apache Arrow
-find_package(Arrow)
+pkg_check_modules(ARROW arrow)
+pkg_check_modules(ARROW_IO arrow-io)
+if (ARROW_FOUND AND ARROW_IO_FOUND)
+  set(ARROW_INCLUDE_DIR ${ARROW_INCLUDE_DIRS})
+  set(ARROW_LIBS ${ARROW_LIBRARY_DIRS})
+
+  set(ARROW_LIB_NAME libarrow)
+  set(ARROW_IO_LIB_NAME libarrow_io)
```

Comment:
Yes.
[`pkg_check_modules`](https://cmake.org/cmake/help/latest/module/FindPkgConfig.html#command:pkg_check_modules) provides `ARROW_LDFLAGS`. It includes all needed for each platform.

But pkg-config isn't used for MSVC normally.

So it's better that we use `link_directories()` and [`target_link_libraries()`](https://cmake.org/cmake/help/latest/command/target_link_libraries.html#command:target_link_libraries) with `arrow` (no `lib` prefix and no `.lib` suffix) instead of `IMPORTED_LOCATION`. `target_link_libraries()` cares about real linker flags (`-larrow` or `arrow.lib`).